src/main/java/request/RequestUri.java

```diff
@@ -0,0 +1,74 @@
+package request;
+
+import annotation.web.RequestMethod;
+import exceptions.MethodNotFoundException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static utils.HttpRequestUtils.*;
+
+
+public class RequestUri {
+    private static final int METHOD_INDEX_IN_LINE = 0;
+    private static final int URI_INDEX_IN_LINE = 1;
+    private RequestMethod requestMethod;
+    private String path;
+    private Map<String, String> params;
+
+    private RequestUri(RequestMethod requestMethod, String path, Map<String, String> params) {
+        this.requestMethod = requestMethod;
+        this.path = path;
+        this.params = params;
+    }
+
+    public static RequestUri from(String line) {
+        String[] splitLine = line.split(" ");
+        Optional<String> queryString = extractParams(splitLine[URI_INDEX_IN_LINE]);
+        if (queryString.isPresent()) {
+            return new RequestUri(getMethodType(splitLine[METHOD_INDEX_IN_LINE])
+                    , extractPath(splitLine[URI_INDEX_IN_LINE])
+                    , requestStringToMap(queryString.get()));
+        }
+        return new RequestUri(getMethodType(splitLine[METHOD_INDEX_IN_LINE])
+                , extractPath(splitLine[URI_INDEX_IN_LINE])
+                , new HashMap<>());
+    }
+
+    private static RequestMethod getMethodType(String line) {
+        String method = line.split(" ")[METHOD_INDEX_IN_LINE];
+        if (method.equals("GET")) {
+            return RequestMethod.GET;
+        }
+        if (method.equals("POST")) {
+            return RequestMethod.POST;
+        }
+        if (method.equals("DELETE")) {
+            return RequestMethod.DELETE;
+        }
+        if (method.equals("PUT")) {
+            return RequestMethod.PUT;
+        }
+        if (method.equals("PATCH")) {
+            return RequestMethod.PATCH;
+        }
+        throw new MethodNotFoundException();
+    }
```

Comment:
enum의 `valueOf` 는 내부적으로 자기 자신 instance를 반환합니다. 
존재하지 않는 값일때는 exception을 던지므로 RequestMethod에는 현재 지원하는 메소드만 지정해 놓고 valueOf를 사용하는건  어떨까요?
```suggestion
private static RequestMethod getMethodType(String line) {
        String method = line.split(" ")[METHOD_INDEX_IN_LINE];
        try {
            return RequestMethod.valueOf(method);
        } catch (IllegalArgumentException ex) {
            throw new MethodNotFoundException();
        }
}